pkg/util/k8sutil/k8sutil.go

```diff
@@ -266,6 +266,11 @@ func NewEtcdPod(m *etcdutil.Member, initialCluster []string, clusterName, state,
 			Volumes: []v1.Volume{
 				{Name: "etcd-data", VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}},
 			},
+			// DNS A record: [m.Name].[clusterName].Namespace.pod.cluster.local.
+			// For example, etcd-0000 in default namesapce will have DNS name
+			// `etcd-0000.etcd.default.pod.cluster.local`.
```

Comment:
@xiang90 We just verified this on k8s 1.6:
```
Given a Pod with the hostname set to “busybox-1” and the
subdomain set to “default-subdomain”, and a headless Service
named “default-subdomain” in the same namespace, the pod will
see it’s own FQDN as 
“busybox-1.default-subdomain.my-namespace.svc.cluster.local”.
```
So we can keep that service around and access the pod by `etcd-0000.etcd.default.svc.cluster.local`.